src/com/ms/silverking/cloud/dht/daemon/storage/convergence/management/ConvergenceControllerBase.java

```diff
@@ -43,272 +43,287 @@
-            }
-            return replicaSyncRequest;
-        } else {
-            return upstreamDependency;
-        }
+      replicaSyncRequest = ReplicaSyncRequest.of(ns, region, newOwner, oldOwner, upstreamDependency);
+      srList.add(replicaSyncRequest);
+      //Log.warningf("Adding %s", replicaSyncRequest);
+      syncController.addAction(replicaSyncRequest);
+      if (++requests - lastDisplayedRequests > requestDisplayInterval) {
+        lastDisplayedRequests = requests;
+        Log.warningAsyncf("Requests: %d\n", requests);
+      }
+      return replicaSyncRequest;
+    } else {
+      return upstreamDependency;
     }
-    
-    public RequestState getRequestState() {
-        if (abandoned || failed) {
-            return RequestState.FAILED;
-        } else {
-            return complete ? RequestState.SUCCEEDED : RequestState.INCOMPLETE;
-        }
-    }
-       
-    public void abandon() {
-        abandoned = true;
-        syncController.abandon();
-        waitForCompletion();
+  }
+
+  public RequestState getRequestState() {
+    if (abandoned || failed) {
+      return RequestState.FAILED;
+    } else {
+      return complete ? RequestState.SUCCEEDED : RequestState.INCOMPLETE;
     }
-    
-    protected void checkForAbandonment() throws ConvergenceException {
-        if (abandoned) {
-            throw new ConvergenceException("Abandoned");
-        }
+  }
+
+  public void abandon() {
+    abandoned = true;
+    syncController.abandon();
+    waitForCompletion();
+  }
+
+  protected void checkForAbandonment() throws ConvergenceException {
+    if (abandoned) {
+      throw new ConvergenceException("Abandoned");
     }
-    
-    protected void setComplete(boolean succeeded) {
-        completionLock.lock();
-        try {
-            failed = !succeeded;
-            complete = true;
-            completionCV.signalAll();
-        } finally {
-            completionLock.unlock();
-        }
+  }
+
+  protected void setComplete(boolean succeeded) {
+    completionLock.lock();
+    try {
+      failed = !succeeded;
+      complete = true;
+      completionCV.signalAll();
+    } finally {
+      completionLock.unlock();
     }
-    
-    protected void waitForCompletion() {
-        completionLock.lock();
+  }
+
+  protected void waitForCompletion() {
+    completionLock.lock();
+    try {
+      while (!complete) {
         try {
-            while (!complete) {
-                try {
-                    completionCV.await();
-                } catch (InterruptedException e) {
-                }
-            }
-        } finally {
-            completionLock.unlock();
-        }
-    }
-    
-    ///////////////////////////////////////////////////
-    
-    protected void ensureUUIDMatches(UUIDBase uuid) {
-        if (!uuid.equals(this.uuid)) {
-            throw new RuntimeException("Unexpected UUID: "+ uuid +" != "+ this.uuid);
+          completionCV.await();
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)